Mathlib/Algebra/CharP/Basic.lean

```diff
@@ -118,6 +118,12 @@ theorem CharP.cast_eq_zero [AddMonoidWithOne R] (p : ℕ) [CharP R p] : (p : R)
   (CharP.cast_eq_zero_iff R p p).2 (dvd_refl p)
 #align char_p.cast_eq_zero CharP.cast_eq_zero
 
+-- See note [no_index around OfNat.ofNat]
+@[simp]
+theorem CharP.ofNat_eq_zero [AddMonoidWithOne R] (p : ℕ) [p.AtLeastTwo] [CharP R p] :
```

Comment:
Does this actually work as intended?
I would think that the `p` if it was a raw nat lit would appear in `CharP` as an `ofNat` too.
I also worry a bit about making the whole LHS of a simp lemma no index, if that ends up matching everything in the discrimination tree I wonder if we would see a performance impact for simp if we have too many of them

Reply:
I worry this lemma might be responsible for the performance hit.

Reply:
Yes, this seems like a bad simplification rule, since this simplification rule tries to match against every subexpression, IIUC. There are also a some other simplification lemmas whose pattern seems too general, and will be pretty expensive, like `ofNat_le`.

But mostly these new simp lemmas are great!

Reply:
Hmm but this lemma is necessary for simp confluence. Should I replace this with a TODO that mentions the no_index library note?

Reply:
simp confluence is important to have, but is it worth it if it makes simp (much) slower? 

I'm wondering if we should start a small list of additional simp sets, that you can opt-in to, e.g. `simp [confluence]` is a simp-set where we try really hard to be confluent.

Reply:
I removed `@[simp]` and added a TODO for now. Hopefully we can make this `simp` once the core bug requiring `no_index` is fixed (https://github.com/leanprover/lean4/issues/2867)